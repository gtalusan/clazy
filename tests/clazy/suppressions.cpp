#include <QtCore/QString>
#include <QtCore/QList>

// clazy:excludeall=foreach

void suppress_qstring_allocation()
{
    QString s = "foo"; // clazy:exclude=qstring-allocations
    if (s == "foo") {} // clazy:exclude=qstring-allocations
    if (s == "foo") {}
}


struct BigTrivial
{
    int a, b, c, d, e;
};

void suppress_foreach()
{
    QList<BigTrivial> list;
    foreach (BigTrivial b, list) { }
}
