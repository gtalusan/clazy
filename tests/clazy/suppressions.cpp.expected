clazy/suppressions.cpp:10:9: warning: QString(const char*) being called [-Wclazy-qstring-allocations]
